Data/ByteString/Streaming/Char8.hs

```diff
@@ -654,7 +654,7 @@ lineSplit !n0 text0 = loop1 0 text0
           let !numNewlines = B.count newline c
               !newCounter = counter + numNewlines
            in if newCounter >= n
-                then case Prelude.drop (n - counter - 1) (B.findIndices (== newline) c) of
+                then case Prelude.drop (n - counter - 1) (B.elemIndices newline c) of
```

Comment:
It might also be worth noting that as of `bytestring 0.11` the the `streaming-bytestring` package will no longer work with GHC 7.x, and so the cabal file needs a patch to cap the `bytestring` version to `< 0.11`.  Something along the lines of:
```
diff --git a/streaming-bytestring.cabal b/streaming-bytestring.cabal
index 435acd4..54bec8c 100644
--- a/streaming-bytestring.cabal
+++ b/streaming-bytestring.cabal
@@ -204,8 +204,10 @@ library
   if impl(ghc < 7.8)
     build-depends: bytestring < 0.10.4.0
                  , bytestring-builder
+  else if impl(ghc < 8.0)
+    build-depends: bytestring >= 0.10.4 && < 0.11
   else
-    build-depends: bytestring >= 0.10.4
+    build-depends: bytestring >= 0.10.4 && < 0.12
 
   if impl(ghc < 8.0)
     build-depends: semigroups
```

Reply:
I ended up reimplementing the function using most of the original logic, but optimizing out the repeated calls to `B.count` by creating a separate internal loop per chunk that "remembers" how many newlines there were originally when it was originally called, and decrements the value by the appropriate amount for each recursive call.

I haven't been able to obtain a practical performance metric of the improvement yet, but the new version compiles and passes the current test-suite on my local system, so it would be possible to compare the performance before and after.